devicetypes/smartthings/aeotec-doorbell-siren-6.src/aeotec-doorbell-siren-6.groovy

```diff
@@ -0,0 +1,245 @@
+	onOffCmd(0x00)
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	def cmds = [
+			encap(zwave.basicV1.basicGet())
+	]
+	sendHubCommand(cmds)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	if (cmd.value == 0) {
+		keepChildOnline()
+        sendEvent(name: "alarm", value: "off", displayed: false)
+	}
+}
+
+def refresh() {
+	ping()
+}
+
+private addChildren(numberOfSounds) {
+	for (def endpoint : 2..numberOfSounds) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+
+			addChildDevice("Aeotec Doorbell Siren Child", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : "$device.displayName Sound $endpoint",
+					isComponent   : true,
+					componentName : "sound$endpoint",
+					componentLabel: "Sound $endpoint"
+			])
+		} catch (Exception e) {
+			log.debug "Excep: ${e} "
+		}
+	}
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint && endpoint > 1) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+		if (zwaveInfo.zw.contains("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+def channelNumber(String dni) {
+	dni[-1] as Integer
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	if (cmd.notificationStatus == 0xFF) {
+		switch (cmd.event) {
+			case 0x09: //TAMPER
+				createEvent([name: "tamper", value: "detected"])
+				break
+			case 0x01: //ON
+				if (state.lastTriggeredSound == 1) {
+					createEvent([name: "alarm", value: "both"])
+				} else {
+					setActiveSound(state.lastTriggeredSound)
+				}
+				break
+			case 0x00: //OFF
+				resetActiveSound()
+				createEvent([name: "tamper", value: "clear"])
+				break
+		}
+	}
+}
+
+def setOnChild(deviceDni) {
+	resetActiveSound()
+	sendHubCommand encap(zwave.basicV1.basicSet(value: 0xFF), channelNumber(deviceDni))
+	state.lastTriggeredSound = channelNumber(deviceDni)
+	setActiveSound(state.lastTriggeredSound)
+}
+
+def setOffChild(deviceDni) {
+	sendHubCommand encap(zwave.basicV1.basicSet(value: 0x00), channelNumber(deviceDni))
+}
+
+def resetActiveSound() {
+	if (state.lastTriggeredSound > 1) {
+		String childDni = "${device.deviceNetworkId}:$state.lastTriggeredSound"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+
+		setOffChild(childDni)
+		child?.sendEvent([name: "chime", value: "off"])
+		child?.sendEvent([name: "alarm", value: "off"])
+	} else {
+		sendHubCommand(onOffCmd(0x00))
+	}
+	sendEvent([name: "alarm", value: "off"])
+	sendEvent([name: "chime", value: "off"])
+}
+
+def setActiveSound(soundId) {
+	String childDni = "${device.deviceNetworkId}:${soundId}"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent([name: "chime", value: "chime"])
+}
+
+def keepChildOnline() {
```

Comment:
I'd probably call this `keepChildrenOnline` since it works on multiple child devices.

Reply:
spaces and name, changed